src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -169,14 +180,61 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
                 // If the user did not specify a version then update the version to resolved version
                 UpdatePackageVersionIfNeeded(restorePreviewResult, packageReferenceArgs, userSpecifiedFrameworks);
 
+                // generate a library dependency with all the metadata like Include, Exlude and SuppressParent
+                var libraryDependency = GenerateLibraryDependency(updatedPackageSpec, packageReferenceArgs, restorePreviewResult.Result.RestoreGraphs, userSpecifiedFrameworks);
+
                 msBuild.AddPackageReferencePerTFM(packageReferenceArgs.ProjectPath,
-                    packageReferenceArgs.PackageDependency,
+                    libraryDependency,
                     compatibleOriginalFrameworks);
             }
 
             return 0;
         }
 
+        private static LibraryDependency GenerateLibraryDependency(
+            PackageSpec project,
+            PackageReferenceArgs packageReferenceArgs,
+            IEnumerable<RestoreTargetGraph> targetGraphs,
+            IEnumerable<NuGetFramework> UserSpecifiedFrameworks)
+        {
+            if (packageReferenceArgs.Frameworks?.Any() == true)
+            {
+                // If the user specified frameworks then we get the flattened graphs  only from the compatible frameworks.
+                var userSpecifiedFrameworkSet = new HashSet<NuGetFramework>(
+                    UserSpecifiedFrameworks,
+                    new NuGetFrameworkFullComparer());
+
+                targetGraphs = targetGraphs
+                    .Where(r => userSpecifiedFrameworkSet.Contains(r.Framework));
+            }
+
+            foreach (var restoreGraph in targetGraphs)
+            {
+                var resolvedEntry = restoreGraph
+                    .Flattened
+                    .SingleOrDefault(p => p.Key.Name.Equals(project.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
The project isn't in the flattened graph, but possibly the removal is only done when writing out the assets file. Should this find a different way to get the library?